modules/c++/math/include/math/Utilities.h

```diff
@@ -61,7 +61,9 @@ inline double square(double val)
  */
 template <typename T> bool isNaN(T value)
 {
-#ifdef HAVE_ISNAN
+#ifdef HAVE_STD_ISNAN
```

Comment:
Including headers inside a namespace does bad things (it'll look like everything included in that header is inside the namespace) - https://stackoverflow.com/questions/2868955/shielding-include-within-namespace-block

So, you'll want to repeat the if/elif up at the top of the file rather than including them down here...